test/fab.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { test, Page, HTML } from "bigtest";
+import { Fab as Interactor, including } from '../src/index';
+import { Fab as MuiFab, Icon as MuiIcon } from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import { render } from './helpers';
+
+const Span = HTML.extend('span').selector('span')
+const SvgIcon = HTML.extend('svg').selector('svg')
+
+const fab = Interactor();
+
+export default test("Fab")
+  .step(Page.visit("/"))
+  .child("can render a floating action button", (test) => test
+    .step("render fab", async() => {
+      await render(<MuiFab>My Fab</MuiFab>)
+    })
+    .assertion(fab.exists())
+    .assertion(fab.has({ text: "My Fab".toUpperCase()}))
+  )
+  .child("renders extended floating action button", (test) => test
+    .step("render extended Fab", async() => {
+      await render(<MuiFab variant="extended">My Fab</MuiFab>)
+    })
+    .assertion(fab.has({ className: including('MuiFab-extended')}))
+  )
+  .child("render Icon with children with right classes", (test) => test
+    .step("render Fab with Icon", async() => {
+      const childClassName = 'child-woof';
+      const iconChild = <MuiIcon data-testid="icon" className={childClassName} />;
+      await render(<MuiFab>{iconChild}</MuiFab>);
+    })
+    .assertion(fab.find(Span({ className: including('child-woof')})).exists())
+  )
+  .child("render Fab with only icon and aria-label", (test) => test
+    .step("render Fab", async() => {
+      await render(
+        <MuiFab color="primary" aria-label="add">
+           <AddIcon />
+        </MuiFab>
+      )
+    })
+    .assertion(fab.has({ ariaLabel: including('add')}))
+    .assertion(fab.has({ svgIcon: true}))
+    // TODO: Fix this assertion fails with:: "ERROR actual.includes is not a function"
+    // .assertion(fab.find(SvgIcon({ className: including('MuiSvgIcon-root') })).exists())
```

Comment:
Do SVG elements necessarily have a class name? or is it in some weird namespace like `svg:className` 🤔

Reply:
@cowboyd I think so. It returns something different when you try to access the className in the DOM. I moved the check into a filter.